runtime/vm/jvminit.c

```diff
@@ -3538,10 +3538,18 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 	if (J2SE_VERSION(vm) >= J2SE_V11) {
 		vm->extendedRuntimeFlags |= (UDATA)J9_EXTENDED_RUNTIME_RESTRICT_IFA; /* Enable zAAP switching for Registered Natives and JVMTI callbacks by default in Java 9 and later. */
 	}
-	vm->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_OSR_SAFE_POINT; /* Enable OSR safe point by default */
-	vm->extendedRuntimeFlags |= (UDATA)J9_EXTENDED_RUNTIME_ENABLE_HCR; /* Enable HCR by default */
 	vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_ENABLE_UTF_CACHE; /* Enable UTF cache by default */
 
+	{
+		IDATA noDynamicAgentLoading = FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXNOENABLEDYNAMICAGENTLOADING, NULL);
+		IDATA dynamicAgentLoading = FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXENABLEDYNAMICAGENTLOADING, NULL);
+		if (noDynamicAgentLoading > dynamicAgentLoading) {
+			vm->extendedRuntimeFlags &= ~(UDATA)(J9_EXTENDED_RUNTIME_OSR_SAFE_POINT | (UDATA) J9_EXTENDED_RUNTIME_ENABLE_HCR);
+		} else {
+			vm->extendedRuntimeFlags |= (J9_EXTENDED_RUNTIME_OSR_SAFE_POINT | (UDATA)J9_EXTENDED_RUNTIME_ENABLE_HCR);
+		}
```

Comment:
A couple of comments:
1. `-XX:[+|-]EnableDynamicAgentLoading` is a new option introduced since Java 9, and should not be applied to Java 8;
2. Is `-XX:[+|-]EnableDynamicAgentLoading` going to take precedence over `-XX:[+|-]EnableHCR` or `-XX:[+|-]OSRSafePoint`? if so, this code block needs to be after processing of `VMOPT_XXENABLEOSRSAFEPOINT` and `VMOPT_XXENABLEHCR`.

Reply:
We should consider if it can apply to Java 8. As long as it doesn't change the default behaviour I don't see why it can't apply.

Reply:
`-XX:[+|-]EnableHCR` and `-XX:[+|-]OSRSafePoint` are Java 8 compatiable, if there is a need to disable such capabilities for Java 8, existing flags can be applied.
If a new flag is introduced, we can apply it to Java 8 if required w/o changing default behaviours.